src/test/java/org/elasticsearch/index/query/MatchAllQueryBuilderTest.java

```diff
@@ -0,0 +1,174 @@
+import org.elasticsearch.threadpool.ThreadPoolModule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.*;
+
+public class MatchAllQueryBuilderTest extends ElasticsearchTestCase {
+
+    private static final String MATCH_ALL_BOOST_1_5 = "{\"match_all\":{\"boost\":1.5}}";
+
+    protected QueryParseContext context;
+
+    protected Injector injector;
+
+    private XContentParser parser;
+
+    private MatchAllQueryBuilder testQuery;
+
+    @Before
+    public void setup() throws IOException {
+        Settings settings = ImmutableSettings.settingsBuilder()
+                .put("path.conf", this.getResourcePath("config"))
+                .put("name", getClass().getName())
+                .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                .build();
+
+        Index index = new Index("test");
+        injector = new ModulesBuilder().add(
+                new EnvironmentModule(new Environment(settings)),
+                new SettingsModule(settings),
+                new ThreadPoolModule(settings),
+                new IndicesQueriesModule(),
+                new ScriptModule(settings),
+                new IndexSettingsModule(index, settings),
+                new IndexCacheModule(settings),
+                new AnalysisModule(settings),
+                new SimilarityModule(settings),
+                new IndexNameModule(index),
+                new IndexQueryParserModule(settings),
+                new FunctionScoreModule(),
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        bind(ClusterService.class).toProvider(Providers.of((ClusterService) null));
+                        bind(CircuitBreakerService.class).to(NoneCircuitBreakerService.class);
+                    }
+                }
+        ).createInjector();
+
+        IndexQueryParserService queryParserService = injector.getInstance(IndexQueryParserService.class);
+        context = new QueryParseContext(index, queryParserService);
+
+        testQuery = createTestQuery();
+        String contentString = createXContent(testQuery).string();
+        parser = XContentFactory.xContent(contentString).createParser(contentString);
+    }
+
+    XContentBuilder createXContent(BaseQueryBuilder query) throws IOException {
+        XContentBuilder content = XContentFactory.jsonBuilder();
+        query.toXContent(content, null);
+        return content;
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        terminate(injector.getInstance(ThreadPool.class));
+    }
+
+    @Test
+    public void testToXContent() throws IOException {
+        XContentBuilder content = createXContent(new MatchAllQueryBuilder().boost(1.5f));
+        assertEquals(content.string(), MATCH_ALL_BOOST_1_5);
```

Comment:
Agreed on the part to avoid comparing to fixed strings. I included this just to have a simple point of reference for the creation of the XContent. We can drop this later I guess. I think what you suggest is what is done in the testFromXContent().